packages/application-shell/src/test-utils/test-utils.spec.js

```diff
@@ -264,3 +274,117 @@ describe('router', () => {
     expect(history.location.pathname).toBe('/foo');
   });
 });
+
+describe('custom render function', () => {
+  describe('wrapper', () => {
+    describe('when provided', () => {
```

Comment:
Nit: I would collapse these describes into `with wrapper` and `without wrapper`.